src/lib.rs

```diff
@@ -327,166 +336,27 @@ pub mod pyclass;
-        }
-    }};
-}
-
-/// A convenient macro to execute a Python code snippet, with some local variables set.
-///
-/// # Panics
-/// This macro internally calls [`Python::run`](struct.Python.html#method.run) and panics
-/// if it returns `Err`, after printing the error to stdout.
-///
-/// # Examples
-/// ```
-/// use pyo3::{prelude::*, py_run, types::PyList};
-/// Python::with_gil(|py| {
-///     let list = PyList::new(py, &[1, 2, 3]);
-///     py_run!(py, list, "assert list == [1, 2, 3]");
-/// });
-/// ```
-///
-/// You can use this macro to test pyfunctions or pyclasses quickly.
-///
-/// ```
-/// use pyo3::{prelude::*, py_run, PyCell};
-/// #[pyclass]
-/// #[derive(Debug)]
-/// struct Time {
-///     hour: u32,
-///     minute: u32,
-///     second: u32,
-/// }
-/// #[pymethods]
-/// impl Time {
-///     fn repl_japanese(&self) -> String {
-///         format!("{}時{}分{}秒", self.hour, self.minute, self.second)
-///     }
-///     #[getter]
-///     fn hour(&self) -> u32 {
-///         self.hour
-///     }
-///     fn as_tuple(&self) -> (u32, u32, u32) {
-///         (self.hour, self.minute, self.second)
-///     }
-/// }
-/// Python::with_gil(|py| {
-///     let time = PyCell::new(py, Time {hour: 8, minute: 43, second: 16}).unwrap();
-///     let time_as_tuple = (8, 43, 16);
-///     py_run!(py, time time_as_tuple, r#"
-///         assert time.hour == 8
-///         assert time.repl_japanese() == "8時43分16秒"
-///         assert time.as_tuple() == time_as_tuple
-///     "#);
-/// });
-/// ```
-///
-/// If you need to prepare the `locals` dict by yourself, you can pass it as `*locals`.
-///
-/// ```
-/// use pyo3::prelude::*;
-/// use pyo3::types::IntoPyDict;
-/// #[pyclass]
-/// struct MyClass {}
-/// #[pymethods]
-/// impl MyClass {
-///     #[new]
-///     fn new() -> Self { MyClass {} }
-/// }
-/// Python::with_gil(|py| {
-///    let locals = [("C", py.get_type::<MyClass>())].into_py_dict(py);
-///    pyo3::py_run!(py, *locals, "c = C()");
-/// });
-/// ```
-///
-/// **Note**
-/// Since this macro is intended to use for testing, it **causes panic** when
-/// [Python::run] returns `Err` internally.
-/// If you need to handle failures, please use [Python::run] directly.
-///
-#[macro_export]
+#[cfg_attr(docsrs, doc(cfg(feature = "macros")))]
 #[cfg(feature = "macros")]
-macro_rules! py_run {
-    ($py:expr, $($val:ident)+, $code:literal) => {{
-        $crate::py_run_impl!($py, $($val)+, $crate::indoc::indoc!($code))
-    }};
-    ($py:expr, $($val:ident)+, $code:expr) => {{
-        $crate::py_run_impl!($py, $($val)+, &$crate::unindent::unindent($code))
-    }};
-    ($py:expr, *$dict:expr, $code:literal) => {{
-        $crate::py_run_impl!($py, *$dict, $crate::indoc::indoc!($code))
-    }};
-    ($py:expr, *$dict:expr, $code:expr) => {{
-        $crate::py_run_impl!($py, *$dict, &$crate::unindent::unindent($code))
-    }};
-}
+pub use pyo3_macros::{pyclass, pyfunction, pymethods, pymodule, pyproto};
```

Comment:
I wonder if we should add a `FromPyObject` import here so that users can write `#[derive(pyo3::FromPyObject)]`? (I think we can also switch to using this in `test_proc_macro_hygiene`.)

And if we add that, I think it's equivalent to

```suggestion
pub use pyo3_macros::*;
```

Reply:
I've imported it explicitly. If I do that with a glob import, this happens, which makes it really confusing where a macro is coming from.

```rust
pub use pyo3_macros::*;

#[macro_use]
mod macros;
```